### ISSUE TITLE
write error: Input/output error

### ISSUE THREAD

**Issue by puterboy** [astrada/google-drive-ocamlfuse#807]
goolge-drive-ocamlfuse has been working for me perfectly for several years.
Starting a couple of days ago, whenever I try to write a file to my google drive, I get the error:
`write error: Input/output error`

Interestingly, it seems like the file is actually written ok - just that I get the error.

Remounting and/or clearing cache did not help.

Did something change in the google drive protocols that is causing this to happen?

This bug seems similar to https://github.com/astrada/google-drive-ocamlfuse/issues/805

I am using version 0.7.27 on Ubuntu 18.04

**Comment by puterboy** [comment]
Also copying (/bin/cp), gives a similar error suggesting the problem is with closing the written file.

`cp test /home/user/gdrive/test`
`cp: failed to close '/home/user/gdrive/test': Input/output error`

Note that deleting (/bin/rm) or touching (/usr/bin/touch) files does not cause errors.

**Comment by StormTrooper** [comment]
I'm also seeing this on Ubuntu 22.10.  Something must have changed on Google's side.  I can read files ok ,but get the error when copying a file.

**Comment by mstubna** [comment]
Also seeing this issue on Ubuntu 20.04 when trying to `rsync` files to google drive.

**Comment by astrada** [comment]
Yes, it seems to be impacting some Drive API users. There is this issue in the Google's tracker: https://issuetracker.google.com/issues/271083853. I'm investigating it.

**Comment by astrada** [comment]
I've uploaded a new version on the beta ppa (https://launchpad.net/~alessandro-strada/+archive/ubuntu/google-drive-ocamlfuse-beta) that uses the "official" path for resumable uploads (https://developers.google.com/drive/api/guides/manage-uploads#resumable) even if the Discovery API (https://www.googleapis.com/discovery/v1/apis/drive/v3/rest) still returns `/resumable/upload/drive/v3/files` as the path for resumable uploads.

**Comment by puterboy** [comment]
For the last 24 hours or so, the original version (not the beta ppa) seems to be working again (I have a process that writes to Google Drive every 15min and I haven't seen any new errors).

Not sure if this has been permanently fixed by Google or not as earlier this week, it was not consistently returning errors until yesterday.

**Comment by StormTrooper** [comment]
Original version is also working again for me.
